### ISSUE TITLE
roadmap for 1.0

### ISSUE THREAD

**Issue by GersonDias** [gabeklein/expressive-mvc#35]
hey @gabeklein, for some time I was thinking about building something like what you are doing here in this project, and why not use an MVC architecture style in my react app. What you did looks amazing, I didn't have the opportunity to play with it, but I had a good time reading the readme and thinking "yes, I have that problem all the time!". Well, my question is, do you think this library is already production-ready? Can you give some thoughts about the roadmap or how to help the project?

**Comment by gabeklein** [comment]
Hi @GersonDias!

Yes, use-controller is really coming along and pretty recently gotten to a home-stretch, as it were. Really, for a project this complicated, it's all about digging deep for edge cases and documenting (at this stage). Been getting some of the weirder race-conditions out of the way; this project's internals are _highly_ sensitive to programmer oversight. Solution has to be rock solid because it's so load-bearing; as a class it can be extended indefinitely.

Where I see 1.0 being ready-to-go is definitely going be test coverage related, and to finish out the API. 

Actually, with regards to help I see "play-testers" being a big asset if anyone's interested! Beyond that, some help documenting will go a long way. Current README is about 6 months old, and a lot of functionality has been both added and refined since then. It's an intense bit of work to come up with thorough use-case.

Anyway, TLDR: More tests, more docs. I see the actual project 98% production-stable already however. Beyond that, just trying to avoid inevitable day-1 patches.

**Comment by gabeklein** [comment]
Closing: this 2021 question predates a full rewrite (monorepo, Component, router, vitest with 100% coverage gates). Pre-1.0 gating decisions are now tracked on the MVC Followups project board rather than a roadmap issue.
